### PR TITLE
feat(web-app): sanitze remote mdx

### LINKED ISSUE #692: Scrub remote MDX content to remove imports and unsupported MDX components
Take the test page a step further by using a file that has unsupported MDX components that need scrubbed out prior to rendering. Same with imports that need removed. See remote MDX library constraints. Consider using React error boundaries to catch MDX rendering errors.

E.g.: https://github.com/carbon-design-system/carbon-website/blob/main/src/pages/components/accordion/usage.mdx

Example of imports that we need to discard: https://github.com/carbon-design-system/carbon-website/blob/main/src/pages/components/dropdown/usage.mdx

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #692

@francinelucca is the real MVP of this PR! 🏆 

#### Changelog

**New**

- Added core Carbon components that are being used across mdx pages
- UnknownComponent
- mdx sanitizer plugin
  - checks components in AST against our list of components, and list of all html components, and replaces them with UnknownComponent if they aren't available.
  - Removes import / export statements
  - Removes any unknown variables
- Error checking for improperly formatted mdx

#### Testing / reviewing
Inside of `/pages/samples/remote-mdx-test.js` replace line `34` with various pages and test. 

```
'/src/pages/components/accordion/style.mdx'
'/src/pages/community/component-index.mdx'
'/src/pages/all-about-carbon/partners.mdx' /
'/src/pages/community/patterns/chatbot/overview.mdx' 
'/src/pages/components/dropdown/code.mdx'
'/src/pages/data-visualization/complex-charts/index.mdx'
'/src/pages/components/date-picker/accessibility.mdx'
```

### CODE REVIEW

**Review by andreancardona — DISMISSED** [review]
🔥 tested locally looks good!

**Review by francinelucca — DISMISSED** [review]
✨

**Comment by jdharvey-ibm** [comment]
@francinelucca and I are going to work on splitting this out into its own package so we can hammer it with unit tests! 🔨

**Comment by alisonjoseph** [comment]
@francinelucca found another page that "breaks"
https://github.com/carbon-design-system/carbon-website/blob/main/src/pages/help/contact-us/index.mdx

> ./pages/assets/contact/index.mdx
Unexpected character `@` (U+0040) in name, expected a name character such as letters, digits, `$`, or `_`; whitespace before attributes; or the end of the tag (note: to create a link in MDX, use `[text](url)`)

```
The Carbon team can also be reached on Twitter:
[@\_carbondesign](https://twitter.com/_carbondesign?lang=en)
```

**Review by alisonjoseph — COMMENTED** [review]
Approved 😉 

✅

File: services/web-app/lib/github.js [hunk 1]
```diff
@@ -58,13 +62,39 @@ export const getRemoteMdxData = async (repoParams, mdxPath) => {
...
   const dirPath = response._links.html.split('/').slice(0, -1).join('/')
 
-  return serialize(usageFileSource, {
-    mdxOptions: {
-      remarkPlugins: [remarkGfm, unwrapImages],
-      rehypePlugins: [[rehypeUrls, mdxImgResolver.bind(null, dirPath)]]
-    },
-    parseFrontmatter: true
-  })
+  let serializedContent = null
```
Comment (jdharvey-ibm) [inline services/web-app/lib/github.js:65]: MDX modification logic should all be moved out of this file and into the sanitizer plugin.

The github file should only be responsible for retrieving data from github.com. It should not peek at (or modify) its contents in any form.

There likely needs to be a separate module or component whose responsibility is to call the github.js functions to obtain the data, set up a remark instance with all of the plugins, and produce sanitized mdx as output.

**Review by jharvey10 — CHANGES_REQUESTED** [review]

**Comment by francinelucca** [comment]
closing: we're starting over

**Pull request closed without merging**